core/src/ByteMatrix.h

```diff
@@ -34,7 +34,7 @@ class ByteMatrix
 
 public:
 	ByteMatrix() = default;
-	ByteMatrix(int width, int height, int val = 0) : _width(width), _height(height), _data(_width * _height, val) { }
+	ByteMatrix(int width, int height, int val = 0) : _width(width), _height(height), _data(_width * _height, static_cast<uint8_t>(val)) { }
```

Comment:
I believe the right fix here would be change the type of val to uint8_t.

I see how unintended narrowing conversions can potentially lead to issues but I believe the strategy of simply making warnings go away by adding explicit casts is counter productive. As can be seen in this first example of those changes. So before adding any of those, I would suggest to look hard if the issue is not actually somewhere else. And as long as the cast is not there one can still ask the compiler for a warning if something is wrong. With the cast the reader and the compiler would have to believe this is truly what the author intended and for this line (and maybe others, too) I believe it is not.

Reply:
Absolutely, my intention is not to force my modifications into the code base but rather to raise a discussion about what was the original intention.

Reply:
OK. Then I'd like to suggest to fix this and the buggy `high` variable and possibly check all other instances where you suggested the static_cast to see if there are more of those (who ever is interested in doing that) but leave rest as is for now.